src/app/application.cpp

```diff
@@ -403,15 +403,63 @@ void Application::processParams(const QStringList &params)
         return;
     }
 #endif
+    BitTorrent::AddTorrentParams torrentParams;
+    bool skipTorrentDialog = false;
 
     foreach (QString param, params) {
         param = param.trimmed();
+
+        // Process strings indicating options specified by the user. Use of
+        // any of these should prevent the "Add new torrent" dialog from
+        // opening:
+
+        if (param.startsWith("@savePath=")) {
+            torrentParams.savePath = param.mid(10);
+            skipTorrentDialog = true;
```

Comment:
Unfortunately the save dialog currently has no facility for passing in torrent options. Meaning if I use command line arguments such as `--save-path`, `--category`, `--start-paused` and so on, those options are not reflected in the save dialog and end up having no effect. They only work when adding a torrent using `BitTorrent::Session::instance()->addTorrent()`.

The thing is that these command line arguments are mainly intended to be used for automating qBittorrent. In those instances it's not desirable to have the save dialog open, hence why I set it up so that specifying any of them prevents it from opening.

I thought adding a `--skip-dialog` option was useful, though, because someone might want to add a torrent, skip the dialog, and use the application settings without overriding anything, which wasn't possible previously.

In order to accomplish what you're talking about, I'd have to make significant changes to `AddNewTorrentDialog` so that it can accept an optional `BitTorrent::AddTorrentParams` parameter and properly pass those settings on when adding the torrent, even the ones that can't be specified by the dialog itself. I'm not very familiar with `AddNewTorrentDialog` so it would take some work and have the potential for introducing new bugs. I'd prefer not to do that unless there's a compelling reason for it.